### PR TITLE
Add API docs plugin to new apps

### PR DESCRIPTION
**Pull request by stefanalund**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/create-app/templates/default-app/packages/app/src/sidebar.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ export const AppSidebar = () => (
     <SidebarDivider />
     {/* Global nav, not org-specific */}
     <SidebarItem icon={HomeIcon} to="./" text="Home" />
+    <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
```
Comment (Fox32) [inline packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:30]: I guess I completely broke this with my changes in #2739?

**Comment by freben** [comment]
Hey! When changing the app setup like this, we are supposed to add a changeset that describes for users how to handle the transition where their app "differs" from the default setup.

See [the recent additions](https://github.com/backstage/backstage/blob/master/packages/create-app/CHANGELOG.md) to create-app for inspiration.

I realize there's a nice docs in here change as well, but this is what we are going with at the moment :)

File: docs/getting-started/configure-app-with-plugins.md [hunk 2]
```diff
@@ -6,7 +6,66 @@ description: Documentation on How Configuring App with plugins
...
+0. Add to the repo:
+
+```bash
+yarn add @backstage/plugin-circleci
+```
+
+1. Add plugin API to your Backstage instance:
+
+```js
+// packages/app/src/api.ts
```
Comment (freben) [inline docs/getting-started/configure-app-with-plugins.md:23]: hm this is `apis.ts` and does not look similar to the actual code in there - double check with the latest app to see what the structure is like now

File: packages/create-app/templates/default-app/packages/app/src/sidebar.tsx (hunk 1 above)
Comment (Rugvip) [reply packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:30]: Wrong # right? 😅

**Comment by freben** [comment]
We misconfigured the tugboat app briefly - if you rebase and push this again, it should build without the tugboat error

File: packages/create-app/templates/default-app/packages/app/src/sidebar.tsx (hunk 1 above)
Comment (freben) [reply packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:30]: @Fox32 You meant to link to another PR than that one right?

@stefanalund Just double checking, this doesn't need a leading slash?

Comment (Fox32) [reply packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:30]: Yeah, I mean #2987 ...

Comment (stefanalund) [reply packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:30]: @freben I agree that it looks odd, but none of the other items has a leading slash: https://github.com/backstage/backstage/blob/master/packages/app/src/components/Root/Root.tsx#L84

File: .changeset/slow-insects-fail.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'example-app': patch
```
Comment (benjdlambert) [inline .changeset/slow-insects-fail.md:2]: example-app doesn't need bumping, you can remove this

**Review by benjdlambert — CHANGES_REQUESTED** [review]

**Comment by stefanalund** [comment]
Is there a way to verify that the new app produces the correct output? Looks like using `npx @backstage/create-app` uses the released version of the script that does not include my local changes.

**Comment by benjdlambert** [comment]
I think you can go into the CLI and do `yarn link` which should create a `symlink` that should point to your local install under `backstage-cli`. Then you can run `backstage-cli create-app` I think outside of the backstage repo

**Review by benjdlambert — APPROVED** [review]

**Review by freben — APPROVED** [review]

**Pull request merged**